climada/hazard/landslide.py

```diff
@@ -188,304 +127,170 @@ def __init__(self):
+        else: 
+            LOGGER.info('No frequency can be derived, no event dates')
+        self.fraction = self.intensity.copy()
 
-        return pixel_height, pixel_width
+        self.check()
 
-    def _intensity_cat_to_prob(self, max_prob):
-        """convert NASA nowcasting categories into occurrence probabilities:
-            highest value category value receives a prob of max_prob, lowest category value
-            receives a prob value of 0"""
-        self.intensity_cat = self.intensity.copy()  # save prob values
-        self.intensity = self.intensity.astype(float)
-        self.intensity.data = self.intensity.data.astype(float)
-        max_value = float(max(self.intensity_cat.data))
-        min_value = float(min(self.intensity_cat.data))
+        if check_plots:
+            self.centroids.plot()
+            self.plot_intensity(0)
 
-        for i, j in zip(*self.intensity.nonzero()):
-            self.intensity[i, j] = float((self.intensity[i, j] - min_value) /
-                                         (max_value - min_value) * max_prob)
+        return self
 
 
-    def _intensity_prob_to_binom(self, n_years):
-        """convert occurrence probabilities in NGI/UNEP landslide hazard map into binary
-        occurrences (yes/no) within a given time frame.
+    def set_ls_prob(self, bbox, path_sourcefile, corr_fact=10e6, n_years=500, 
+                    check_plots=False):
+        """
+        Set probabilistic landslide hazard (fraction, intensity and
+        frequency) for a defined bounding box and time period from a raster.
+        The hazard data for which this function is explicitly written
+        is readily provided by UNEP & the Norwegian Geotechnical Institute
+        (NGI), and can be downloaded and unzipped from
+        https://preview.grid.unep.ch/index.php?preview=data&events=landslides&evcat=2&lang=eng
+        for precipitation-triggered landslide and from
+        https://preview.grid.unep.ch/index.php?preview=data&events=landslides&evcat=1&lang=eng
+        for earthquake-triggered landslides.
+        It works of course with any similar raster file.
+        Original data is given in expected annual probability and percentage
+        of pixel of occurrence of a potentially destructive landslide event
+        x 1000000 (so be sure to adjust this by setting the correction factor).
+        More details can be found in the landslide tutorial and under above-
+        mentioned links.
+
+        The annual occurrence probabilites are sampled from a binomial 
+        distribution; intensity takes a binary value (0 - no
+        ls occurrence; 1 - ls occurrence) and
+        fraction stores the actual the occurrence count (0 to n) per grid cell.
+        Frequency is occurrence count / n_years.
+
+        Impact functions, since they act on the intensity, should hence be in
+        the form of a step function,
+        defining impact for intensity 0 and (close to) 1.
```

Comment:
Why? Could I not want to do an impact function which gives different damages depending on how many occurences there were? (This would require the fraction and intensity to be swapped)

Reply:
mmh no, this should be included anyways, because impact = intensity*MDD*fraction

Reply:
Yes, but only as direct addition. This means 2 events always imply twice the damage, which is not necessarily meaningful / desired.